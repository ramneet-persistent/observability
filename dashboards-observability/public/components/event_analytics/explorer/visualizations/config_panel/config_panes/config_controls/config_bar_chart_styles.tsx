/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useMemo, useCallback, Fragment } from 'react';
import { EuiAccordion, EuiSpacer } from '@elastic/eui';
import { get } from 'lodash';
import { ButtonGroupItem } from './config_button_group';
import { IConfigPanelOptionSection } from '../../../../../../../../common/types/explorer';
import {
  SLIDER_MIN_VALUE,
  SLIDER_MAX_VALUE,
  SLIDER_DEFAULT_STEP,
} from '../../../../../../../../common/constants/shared';
import { ConfigChartOptionsEnum } from '../../../../../../../../common/constants/explorer';

export const ConfigBarChartStyles = ({
  visualizations,
  schemas,
  vizState,
  handleConfigChange,
  sectionName,
  sectionId = 'chartStyles',
}: any) => {
  const { data } = visualizations;
  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;

  const handleConfigurationChange = useCallback(
    (stateFieldName) => {
      return (changes) => {
        handleConfigChange({
          ...vizState,
          [stateFieldName]: changes,
        });
      };
    },
    [handleConfigChange, vizState]
  );

  /* To update the schema options based on current style mode selection */
  const currentSchemas = useMemo(() => {
    if (vizState?.orientation === 'h') {
      return schemas.filter((schema: IConfigPanelOptionSection) => schema.mapTo !== 'labelAngle');
    }
    return schemas;
  }, [vizState]);

  const dimensions = useMemo(
    () =>
      currentSchemas.map((schema: IConfigPanelOptionSection, index: number) => {
        let params = {
          title: schema.name,
          vizState,
          ...schema.props,
        };
        const DimensionComponent = schema.component || ButtonGroupItem;

        const createDimensionComponent = (dimProps) => (
          <Fragment key={`viz-series-${index}`}>
            <DimensionComponent {...dimProps} />
            <EuiSpacer size="s" />
          </Fragment>
        );
        switch (schema.eleType) {
          case ConfigChartOptionsEnum.buttons:
            params = {
              ...params,
              legend: schema.name,
              groupOptions: schema?.props?.options.map((btn: { name: string }) => ({
                ...btn,
                label: btn.name,
              })),
              idSelected: vizState[schema.mapTo] || schema?.props?.defaultSelections[0]?.id,
              handleButtonChange: handleConfigurationChange(schema.mapTo),
            };
            return createDimensionComponent(params);
          case ConfigChartOptionsEnum.input:
            params = {
              title: schema.name,
              currentValue: vizState[schema.mapTo] || '',
              numValue: vizState[schema.mapTo] || '',
              handleInputChange: handleConfigurationChange(schema.mapTo),
              vizState,
              ...schema.props,
            };
            return createDimensionComponent(params);

          case ConfigChartOptionsEnum.slider:
            params = {
              ...params,
              minRange:
                typeof get(schema, 'props.min') === undefined
                  ? SLIDER_MIN_VALUE
                  : get(schema, 'props.min'),
              maxRange:
                typeof get(schema, 'props.max') === undefined
                  ? SLIDER_MAX_VALUE
                  : get(schema, 'props.max'),
              step: schema?.props?.step || SLIDER_DEFAULT_STEP,
              currentRange: vizState[schema.mapTo] || schema?.defaultState,
              ticks: schema?.props?.ticks,
              showTicks: schema?.props?.showTicks || false,
              handleSliderChange: handleConfigurationChange(schema.mapTo),
            };
            return createDimensionComponent(params);

          default:
            params = {
              ...params,
              paddingTitle: schema.name,
              advancedTitle: 'advancedTitle',
              dropdownList:
                schema?.options?.map((option) => ({ ...option })) ||
                fields.map((item) => ({ ...item })),
              onSelectChange: handleConfigurationChange(schema.mapTo),
              isSingleSelection: schema.isSingleSelection,
              selectedAxis: vizState[schema.mapTo] || schema.defaultState,
            };
            return createDimensionComponent(params);
        }
      }),
    [currentSchemas, vizState, handleConfigurationChange]
  );

  return (
    <EuiAccordion
      initialIsOpen
      id={`configPanel__${sectionId}`}
      buttonContent={sectionName}
      paddingSize="s"
    >
      {dimensions}
    </EuiAccordion>
  );
};
