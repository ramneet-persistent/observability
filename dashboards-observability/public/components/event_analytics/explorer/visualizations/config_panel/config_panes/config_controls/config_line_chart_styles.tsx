/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useMemo, useCallback, Fragment } from 'react';
import { EuiAccordion, EuiSpacer } from '@elastic/eui';
import { get } from 'lodash';
import { ButtonGroupItem } from './config_button_group';
import { IConfigPanelOptionSection } from '../../../../../../../../common/types/explorer';
import {
  visChartTypes,
  SLIDER_MIN_VALUE,
  SLIDER_MAX_VALUE,
  SLIDER_DEFAULT_STEP,
} from '../../../../../../../../common/constants/shared';
import { ConfigChartOptionsEnum } from '../../../../../../../../common/constants/explorer';

export const ConfigLineChartStyles = ({
  visualizations,
  schemas,
  vizState,
  handleConfigChange,
  sectionName,
  sectionId = 'chartStyles',
}: any) => {
  const { data } = visualizations;
  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;

  const handleConfigurationChange = useCallback(
    (stateFiledName) => {
      return (changes) => {
        handleConfigChange({
          ...vizState,
          [stateFiledName]: changes,
        });
      };
    },
    [handleConfigChange, vizState]
  );

  /* To update the schema options based on current style mode selection */
  const currentSchemas = useMemo(() => {
    if (vizState?.style) {
      switch (vizState.style) {
        case 'lines':
          return schemas.filter(
            (schema: IConfigPanelOptionSection) => schema.mapTo !== 'markerSize'
          );
        case 'bar':
          return schemas.filter(
            (schema: IConfigPanelOptionSection) =>
              !['interpolation', 'markerSize'].includes(schema.mapTo)
          );
        case 'markers':
          return schemas.filter((schema: IConfigPanelOptionSection) =>
            ['style', 'markerSize'].includes(schema.mapTo)
          );
        case 'lines+markers':
          return schemas.filter(
            (schema: IConfigPanelOptionSection) => schema.mapTo !== 'interpolation'
          );
      }
    } else if (visualizations?.vis?.name === visChartTypes.Scatter) {
      return schemas.filter((schema: IConfigPanelOptionSection) =>
        ['style', 'markerSize'].includes(schema.mapTo)
      );
    } else {
      return schemas.filter((schema: IConfigPanelOptionSection) => schema.mapTo !== 'markerSize');
    }
  }, [vizState]);

  const dimensions = useMemo(
    () =>
      currentSchemas &&
      currentSchemas.map((schema: IConfigPanelOptionSection, index: number) => {
        const DimensionComponent = schema.component || ButtonGroupItem;
        let params = {
          title: schema.name,
          vizState,
          ...schema.props,
        };

        switch (schema.eleType) {
          case ConfigChartOptionsEnum.buttons:
            params = {
              ...params,
              legend: schema.name,
              groupOptions: schema?.props?.options.map((btn: { name: string }) => ({
                ...btn,
                label: btn.name,
              })),
              idSelected: vizState[schema.mapTo] || schema?.props?.defaultSelections[0]?.id,
              handleButtonChange: handleConfigurationChange(schema.mapTo),
            };
            break;
          case ConfigChartOptionsEnum.slider:
            params = {
              ...params,
              minRange:
                typeof get(schema, 'props.min') === undefined
                  ? SLIDER_MIN_VALUE
                  : get(schema, 'props.min'),
              maxRange:
                typeof get(schema, 'props.max') === undefined
                  ? SLIDER_MAX_VALUE
                  : get(schema, 'props.max'),
              step: schema?.props?.step || SLIDER_DEFAULT_STEP,
              currentRange: vizState[schema.mapTo] || schema?.defaultState,
              ticks: schema?.props?.ticks,
              showTicks: schema?.props?.showTicks || false,
              handleSliderChange: handleConfigurationChange(schema.mapTo),
            };
            break;
          case ConfigChartOptionsEnum.input:
            params = {
              ...params,
              numValue: vizState[schema.mapTo] || '',
              handleInputChange: handleConfigurationChange(schema.mapTo),
            };
            break;

          default:
            params = {
              ...params,
              paddingTitle: schema.name,
              advancedTitle: 'advancedTitle',
              dropdownList:
                schema?.options?.map((option) => ({ ...option })) ||
                fields.map((item) => ({ ...item })),
              onSelectChange: handleConfigurationChange(schema.mapTo),
              isSingleSelection: schema.isSingleSelection,
              selectedAxis: vizState[schema.mapTo] || schema.defaultState,
            };
        }

        return (
          <Fragment key={`viz-series-${index}`}>
            <DimensionComponent {...params} />
            <EuiSpacer size="s" />
          </Fragment>
        );
      }),
    [currentSchemas, vizState, handleConfigurationChange]
  );

  return (
    <EuiAccordion
      initialIsOpen
      id={`configPanel__${sectionId}`}
      buttonContent={sectionName}
      paddingSize="s"
    >
      {dimensions}
    </EuiAccordion>
  );
};
