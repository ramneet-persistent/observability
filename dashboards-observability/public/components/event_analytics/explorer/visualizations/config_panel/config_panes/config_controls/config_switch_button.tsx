/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiSwitch, EuiFormRow, EuiSpacer } from '@elastic/eui';

interface SwitchButtonProps {
  currentValue: boolean;
  title: string;
  onToggle: (value: boolean) => void;
}
export const SwitchButton = ({
  currentValue,
  onToggle,
  title,
}: SwitchButtonProps) => {
  return (
    <>
      {/* <EuiFormRow display="columnCompressedSwitch"  label={title}> */}
      <EuiSpacer />
      <EuiSwitch
        label={title}
        checked={currentValue}
        onChange={(e) => onToggle(e.target.checked)}
      />
      {/* </EuiFormRow> */}
    </>
  );
};
