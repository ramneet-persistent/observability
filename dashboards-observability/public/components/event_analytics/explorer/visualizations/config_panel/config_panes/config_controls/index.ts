/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { ConfigPanelOptions } from './config_panel_options';
export { ConfigValueOptions } from './config_value_options';
export { ConfigChartOptions } from './config_chart_options';
export { ConfigDataLinks } from './config_data_links';
export { ConfigThresholds } from './config_thresholds';
export { ConfigText } from './config_text';
export { ConfigGaugeValueOptions } from './config_gauge_options';
export { ConfigInputField } from "./config_input_field";
export { ConfigGridColumns } from "./config_grid_columns"
export { ConfigColorTheme } from './config_color_theme';

