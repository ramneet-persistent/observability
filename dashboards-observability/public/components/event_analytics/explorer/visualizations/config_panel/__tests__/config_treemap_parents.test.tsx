/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { ConfigTreemapParentFields } from '../config_panes/config_controls/config_treemap_parents';
import { TEST_VISUALIZATIONS_DATA } from '../../../../../../../test/event_analytics_constants';

describe('ConfigTreemapParentFields component', () => {
  configure({ adapter: new Adapter() });

  it('Renders ConfigTreemapParentFields with empty data', async () => {
    const wrapper = mount(
      <ConfigTreemapParentFields
        selectedAxis={[]}
        setSelectedParentItem={jest.fn()}
        handleUpdateParentFields={jest.fn()}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('Renders ConfigTreemapParentFields with data', async () => {
    const wrapper = mount(
      <ConfigTreemapParentFields
        selectedAxis={[
          {
            name: 'spectrum',
            label: 'spectrum',
            type: 'spectrum',
          },
        ]}
        setSelectedParentItem={jest.fn()}
        handleUpdateParentFields={jest.fn()}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
