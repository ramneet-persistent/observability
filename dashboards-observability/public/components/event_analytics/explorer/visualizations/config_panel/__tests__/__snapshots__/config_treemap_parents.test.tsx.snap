// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigTreemapParentFields component Renders ConfigTreemapParentFields with empty data 1`] = `
<ConfigTreemapParentFields
  handleUpdateParentFields={[MockFunction]}
  selectedAxis={Array []}
  setSelectedParentItem={[MockFunction]}
>
  <EuiSpacer
    size="s"
  >
    <div
      className="euiSpacer euiSpacer--s"
    />
  </EuiSpacer>
  <EuiPanel
    className="panelItem_button"
    grow={true}
  >
    <div
      className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow panelItem_button"
    >
      <EuiText
        size="s"
      >
        <div
          className="euiText euiText--small"
        >
          Add Parent
        </div>
      </EuiText>
      <EuiButtonIcon
        aria-label="clear-field"
        color="primary"
        iconSize="s"
        iconType="plusInCircle"
        onClick={[Function]}
      >
        <button
          aria-label="clear-field"
          className="euiButtonIcon euiButtonIcon--primary euiButtonIcon--empty euiButtonIcon--xSmall"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <EuiIcon
            aria-hidden="true"
            className="euiButtonIcon__icon"
            color="inherit"
            size="s"
            type="plusInCircle"
          >
            <EuiIconEmpty
              aria-hidden={true}
              className="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
              focusable="false"
              role="img"
              style={null}
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              />
            </EuiIconEmpty>
          </EuiIcon>
        </button>
      </EuiButtonIcon>
    </div>
  </EuiPanel>
</ConfigTreemapParentFields>
`;
